src/components/HelpScoutBeacon/BeaconHeadScripts.js

```diff
@@ -1,67 +1,72 @@
-import React, { useEffect } from 'react'
+import React, { useCallback, useEffect, useState } from 'react'
 import PropTypes from 'prop-types'
-import { Helmet } from 'react-helmet'
-import { noop, GU } from '../../utils'
+import { GU } from '../../utils'
+import { createGlobalStyle } from 'styled-components'
 
 const BEACON_EMBED =
   '!function(e,t,n){function a(){var e=t.getElementsByTagName("script")[0],n=t.createElement("script");n.type="text/javascript",n.async=!0,n.src="https://beacon-v2.helpscout.net",e.parentNode.insertBefore(n,e)}if(e.Beacon=n=function(t,n,a){e.Beacon.readyQueue.push({method:t,options:n,data:a})},n.readyQueue=[],"complete"===t.readyState)return a();e.attachEvent?e.attachEvent("onload",a):e.addEventListener("load",a,!1)}(window,document,window.Beacon||function(){});'
-const HELPSCOUT_ID = "'163e0284-762b-4e2d-b3b3-70a73a7e6c9f'"
-const BEACON_INIT = "window.Beacon('init'," + HELPSCOUT_ID + ')'
+const HELPSCOUT_ID = '163e0284-762b-4e2d-b3b3-70a73a7e6c9f'
 
-const BeaconHeadScripts = React.memo(({ optedIn, onReady }) => {
+function useHelpScoutBeacon(optedIn) {
+  const [beaconInit, setBeaconInit] = useState(false)
+
+  // Load the script if it doesn’t exist yet and optedIn
+  // is true, then set the value of Beacon.
   useEffect(() => {
-    let timeout = null
-    if (optedIn) {
-      ;(function isBeaconReady() {
-        if (window.Beacon) {
-          onReady()
-          return
-        }
-        timeout = setTimeout(isBeaconReady, 100)
-      })()
+    let script
 
-      return () => clearTimeout(timeout)
+    if (optedIn && !beaconInit) {
+      if (!window.Beacon) {
+        script = document.createElement('script')
+        script.innerHTML = BEACON_EMBED
+        document.body.appendChild(script)
+      }
+      window.Beacon('init', HELPSCOUT_ID)
+      setBeaconInit(true)
     }
-  }, [optedIn, onReady])
 
-  if (!optedIn) {
-    return null
-  }
+    return () => script.remove()
+  }, [optedIn, beaconInit])
 
-  return (
-    <Helmet>
-      <script type="text/javascript">{BEACON_EMBED}</script>
-      <script type="text/javascript">{BEACON_INIT}</script>
-      <style>
-        {`
-          .BeaconFabButtonFrame,
-          #beacon-container .Beacon div:first-of-type {
-            display: none !important;
-          }
-          @media (min-width: 768px) {
-            #beacon-container .BeaconContainer {
-              height: 600px !important;
-              width: 350px !important;
-              top: unset !important;
-              left: unset !important;
-              bottom: 80px !important;
-              right: ${3 * GU}px !important;
-            }
-          }
-        `}
-      </style>
-    </Helmet>
+  const beacon = useCallback(
+    (...params) => {
+      if (window.Beacon && optedIn && beaconInit) {
```

Comment:
What happens when this function is called but `window.Beacon` does not exist is the calls gets lost, should we be queueing them? or at least sending them to the future via a `setTimeout`?

Reply:
I need to check again, but IIRC this `window.Beacon` in the condition is only here as a safety measure because we don’t control what the `HelpScout` script does with the object.

The `window.Beacon` object should exist as soon as `optedIn` changes to be `true`, and `BEACON_EMBED` is doing exactly this:

1. Creating the `window.Beacon` function.
2. Queuing calls to `window.Beacon`.
3. Injecting the script (that uses the queue).

As a note, the reason why it there is a wrapping function here (instead of exposing `window.Beacon` directly) is because HelpScout replaces it in a totally unpredictable way.

Reply:
> The window.Beacon object should exist as soon as optedIn

This is not in synch right? So, maybe there is some delay before the `window.Beacon` object exists.